### ISSUE TITLE
E1111: doesn't respect async functions

### ISSUE THREAD

**Issue by xen** [PyCQA/pylint#2378]
### Steps to reproduce
1. The code I have (`tt.py`):

```python
import asyncio


async def coroutine():
    print('in coroutine')


event_loop = asyncio.get_event_loop()
try:
    coro = coroutine()
    print(f"{type(coro)}: {coro}")
    event_loop.run_until_complete(coro)
finally:
    event_loop.close()
```

2. The versions I use:

```
$ python -V
Python 3.6.5
$ pylint  --version
Using config file /Users/xen/.pylintrc
pylint 1.9.2,
astroid 1.6.5
Python 3.6.5 (default, Jul  9 2018, 21:21:30)
[GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)]
```

3. The output I get:

```
$ pylint tt.py
Using config file /Users/xen/.pylintrc
************* Module tt
E: 10, 4: Assigning to function call which doesn't return (assignment-from-no-return)


Report
======
9 statements analysed.

Statistics by type
------------------

+---------+-------+-----------+-----------+------------+---------+
|type     |number |old number |difference |%documented |%badname |
+=========+=======+===========+===========+============+=========+
|module   |1      |1          |=          |0.00        |0.00     |
+---------+-------+-----------+-----------+------------+---------+
|class    |0      |0          |=          |0           |0        |
+---------+-------+-----------+-----------+------------+---------+
|method   |0      |0          |=          |0           |0        |
+---------+-------+-----------+-----------+------------+---------+
|function |1      |1          |=          |0.00        |0.00     |
+---------+-------+-----------+-----------+------------+---------+



Raw metrics
-----------

+----------+-------+------+---------+-----------+
|type      |number |%     |previous |difference |
+==========+=======+======+=========+===========+
|code      |12     |75.00 |NC       |NC         |
+----------+-------+------+---------+-----------+
|docstring |0      |0.00  |NC       |NC         |
+----------+-------+------+---------+-----------+
|comment   |0      |0.00  |NC       |NC         |
+----------+-------+------+---------+-----------+
|empty     |4      |25.00 |NC       |NC         |
+----------+-------+------+---------+-----------+



Duplication
-----------

+-------------------------+------+---------+-----------+
|                         |now   |previous |difference |
+=========================+======+=========+===========+
|nb duplicated lines      |0     |0        |=          |
+-------------------------+------+---------+-----------+
|percent duplicated lines |0.000 |0.000    |=          |
+-------------------------+------+---------+-----------+



Messages by category
--------------------

+-----------+-------+---------+-----------+
|type       |number |previous |difference |
+===========+=======+=========+===========+
|convention |0      |0        |=          |
+-----------+-------+---------+-----------+
|refactor   |0      |0        |=          |
+-----------+-------+---------+-----------+
|warning    |0      |0        |=          |
+-----------+-------+---------+-----------+
|error      |1      |1        |=          |
+-----------+-------+---------+-----------+



Messages
--------

+--------------------------+------------+
|message id                |occurrences |
+==========================+============+
|assignment-from-no-return |1           |
+--------------------------+------------+




------------------------------------------------------------------
Your code has been rated at 4.44/10 (previous run: 4.44/10, +0.00)
```

### Expected behavior

I expect that I will not get the error `E1111` in the report.

**Comment by xen** [comment]
Sorry, for some reasons I have an outdated version of `pylint`. I've updated to 2.x and error is gone.
